src/box/iproto.cc

```diff
@@ -1828,13 +1832,16 @@ tx_process_begin(struct cmsg *m)
 {
 	struct iproto_msg *msg = tx_accept_msg(m);
 	struct obuf *out;
+	double timeout;
 
 	if (tx_check_schema(msg->header.schema_version))
 		goto error;
 
 	if (box_txn_begin() != 0)
 		goto error;
 
+	timeout = (msg->txn.timeout != 0 ? msg->txn.timeout : txn_timeout);
+	box_txn_set_timeout(timeout);
```

Comment:
No! We discussed about it before, by default ALL transactions has timeout == TIMEOUT_INFINITY and box.cfg.txn_timeout does not affect this behaviour! (It's necessary because we don't wan't to affect single statement transactions, applier transactions and so on). We use box_txn_set_timeout to specify timeout in places where we wan't to  specify timeout for transactions: in iproto for streams and in box.begin. So we should call this function with default timeout or with timeout, that was passed to the function.

Reply:
box_txn_begin() is a part of the public API. It must respect the default timeout.

Let's implement box_txn_begin like this:

```c
void box_txn_begin(void)
{
    struct txn *txn = txn_begin();
    txn_set_timeout(txn, box_default_txn_timeout);
}
```

Then box_txn_begin() will use the default timeout while txn_begin() won't. I think you suggested something like this initially, and after looking at the patch I agree that this would probably be the best option.

Then the default timeout should live in box.cc, where you initially placed it (please discard my comment where I asked you to move to txn.cc, because I thought it would be used by txn_begin() by default). And it should be named accordingly, e.g. box_default_txn_timeout or box_txn_timeout_default.